### PR TITLE
feat: Add a speedier script tag for prerendered redirects

### LINKED ISSUE #9907: Redirect from prerendered pages with `<script>`
### Describe the problem

When prerendering, if we encounter a redirect it gets baked out as an HTML file with a `<meta>` refresh tag:

https://github.com/sveltejs/kit/blob/bc70b4e636fcbd9593356996bf737e014ff8c238/packages/kit/src/core/postbuild/prerender.js#L341-L344

[Turns out](https://twitter.com/cramforce/status/1646509315073929216) (h/t @cramforce) it's actually quicker to use a `<script>`:

```html
<script>onload=() => location.href = "${url}"</script>
```

(Do we even need the `onload`? Would it be quicker still without it?)

### Describe the proposed solution

Do both — `<script>` for speed, `<meta>` for reliability.

### Alternatives considered

_No response_

### Importance

nice to have

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by tcc-sejohnson**
Closes #9907.

TODO:
- [ ] I'm not sure I landed on the right solution for encoding URIs. It passes the tests... but I'm not sure that means it's good.

(I also deleted some empty +page.svelte files, which used to be required but now are not)

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### CODE REVIEW

**Comment by tcc-sejohnson** [comment]
Looks like I missed a few tests; will have to go back and fix 'em

File: .changeset/hungry-rocks-hunt.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@sveltejs/kit': minor
+---
+
+feat: add a speedier script for prerendered redirects
```
Comment (dummdidumm) [inline .changeset/hungry-rocks-hunt.md:5]: Not sure this warrants a minor 😄 
```suggestion
'@sveltejs/kit': patch
---

perf: add a script to prerendered redirects for faster redirects
```

Comment (tcc-sejohnson) [reply .changeset/hungry-rocks-hunt.md:5]: Yeah, I think you're right

**Review by Rich-Harris — APPROVED** [review]
i feel good about the encoding, but this has historically been a blind spot of mine so i'd be happy to wait for others to weigh in (@Conduitry is particularly adept at heading off string encoding bugs)

**Comment by Conduitry** [comment]
Hm. What specifically are we trying to handle by doing `encodeURI(decodeURIComponent(...))`? I'm worried about that losing the encoding on some characters that we actually need it for.

Since `"` and `\` and `<` are all encoded by `encodeURI`, I think we're safe from injection attacks here. But I am still worried about ending up on the wrong URL. For example, a `%40` in the original URL would end up a `@` in the `location.href=` string.

Can we instead do a straight string escaping thing of `location` like we do for the values that get serialized into the inline `<script>` tag that starts the app on regular pages? Note that this is _not_ just `JSON.stringify()` because there are some additional characters that need to be escaped to be safe. Even using `devalue` should work if we don't have another function for this hanging around anywhere, although it would definitely be overkill. It being overkill might not matter so much, since this is done at build time anyway, not runtime.

**Comment by tcc-sejohnson** [comment]
Thanks @Conduitry -- the reason we were doing a decode/encode is to avoid double-encoding. That being said, I think you're right about just string-escaping it. I'm not going to have time to do this today or tomorrow, so I'll probably get it done on Monday.

**Comment by tcc-sejohnson** [comment]
@Conduitry -- mind taking a look now that we're using `devalue.uneval`?

File: .changeset/hungry-rocks-hunt.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@sveltejs/kit': patch
+---
+
+perf: add a script to prerendered redirects for faster redirects
```
Comment (Conduitry) [inline .changeset/hungry-rocks-hunt.md:5]:
```suggestion
perf: add `<script>` to prerendered redirects for faster redirects
```

**Pull request merged**